index.bs

```diff
@@ -796,38 +796,46 @@ partial interface MLContext {
+    1. If invoking the <a>validate graph resources</a> algorithm given |outputs| and |graph|.{{MLGraph/[[outputDescriptors]]}} returns `false`, then throw a "{{DataError}}" {{DOMException}} and stop.
+    1. Invoke <a>execute graph</a> given |graph|, |inputs| and |outputs|.
+    1. If that throws an error, re-throw the error and stop.
+    1. Return {{undefined}}.
+
+<div class=validusage>
+To <dfn lt="validate graph resources">validate graph resources</dfn>, given |resources| and |descriptors|, run the following steps:
+    1. [=Assert=]: the type of |resources| is {{MLNamedArrayBufferViews}}.
+    1. For each [=record=] <|key|, |value|> of |resources|:
+        1. If |descriptors|[|key|] does not [=map/exist=], return `false`.
+        1. [=Assert=]: the type of |value| is {{ArrayBufferView}}.
+        1. If the [=element type=] of |value| does not match |descriptors|[|key|].{{MLOperandDescriptor/type}} according to [this table](#appendices-mloperandtype-arraybufferview-compatibility), return `false`.
+        1. If |value|.\[[ByteLength]] is not equal to [=byte length=] of |descriptors|[|key|], return `false`.
+    1. Return `true`.
+</div>
+
+To <dfn lt="execute graph">execute graph</dfn>, given |graph|, |inputs| and |outputs|, run the following steps:
+    1. [=Assert=]: the type of |inputs| is {{MLNamedArrayBufferViews}}.
+    1. Let |inputResources| denote the input resources of |graph|.{{MLGraph/[[implementation]]}}.
+    1. For each <|key|, |inputValue|> of |inputs|:
+        1. Let |inputDescriptor| be |graph|.{{MLGraph/[[inputDescriptors]]}}[|key|].
+        1. Let |inputTensor| be a new tensor for |graph|.{{MLGraph/[[implementation]]}} as follows:
+            1. Set the data type of |inputTensor| to the one that matches the [=element type=] of |inputValue|.
+            1. Set the dimensions of |inputTensor| to |inputDescriptor|.{{MLOperandDescriptor/dimensions}}.
+            1. Set the values of elements in |inputTensor| to the values of elements in |inputValue|.
+        1. Request the underlying implementation of |graph| to bind |inputResources|[|key|] to |inputTensor|.
+    1. [=Assert=]: the type of |outputs| is {{MLNamedArrayBufferViews}}.
+    1. Let |results| be the result of invoking |graph|.{{MLGraph/[[implementation]]}} to compute the output of |graph| given |inputResources|. This may involve multiple steps and compute requests.
+        1. If that returns an error, throw an "{{OperationError}}" {{DOMException}} and stop.
+    1. [=Assert=]: the type of |results| is {{MLNamedArrayBufferViews}}.
+    1. For each <|key|, |outputValue|> of |outputs|:
+        1. Let |outputDesc| be |graph|.{{MLGraph/[[outputDescriptors]]}}[|key|].
+        1. Issue a compute request to |graph|.{{MLGraph/[[implementation]]}} to provide the output associated with |key| given |inputResources|[|key|] and stored in |resultTensor|.
```

Comment:
Should `|inputResources|[|key|]` be `|inputResources|`? The `|key|` is for output.